wgpu-native/src/command/mod.rs

```diff
@@ -313,22 +394,41 @@ pub fn command_encoder_begin_render_pass(
     let render_pass = match render_pass_cache.entry(rp_key.clone()) {
         Entry::Occupied(e) => e.into_mut(),
         Entry::Vacant(e) => {
+            let attachment_unused: u32 = !0; // TODO: Get from ash or expose in gfx-hal. Very important this remains a u32.
+
             let color_ids = [
                 (0, hal::image::Layout::ColorAttachmentOptimal),
                 (1, hal::image::Layout::ColorAttachmentOptimal),
                 (2, hal::image::Layout::ColorAttachmentOptimal),
                 (3, hal::image::Layout::ColorAttachmentOptimal),
             ];
+
+            let mut resolve_ids = ArrayVec::<[_; crate::device::MAX_COLOR_TARGETS]>::new();
+            let mut attachment_index = color_attachments.len();
+            let resolve_target_count = color_attachments.iter().filter(|at| at.resolve_target != ptr::null()).count();
+            if resolve_target_count != 0 {
+                for (i, at) in color_attachments.iter().enumerate() {
+                    resolve_ids.push(if at.resolve_target == ptr::null() {
+                        (attachment_unused as usize, hal::image::Layout::ColorAttachmentOptimal)
+                    } else {
+                        let sample_count_check = view_guard[color_attachments[i].attachment].samples;
+                        assert!(sample_count_check > 1, "RenderPassColorAttachmentDescriptor with a resolve_target must have an attachment with sample_count > 1");
+                        (attachment_index, hal::image::Layout::ColorAttachmentOptimal)
+                    });
+                    attachment_index += 1;
```

Comment:
this line seems wrong, should it be moved under `else`?

Reply:
Good catch!